Framework/CurveFitting/src/Algorithms/LeBailFit.cpp

```diff
@@ -1508,10 +1511,10 @@ void LeBailFit::execRandomWalkMinimizer(size_t maxcycles,
   calculateDiffractionPattern(vecX, vecPurePeak, false, false, vecBkgd,
                               vecCalPurePeaks, finalR);
 
-  MantidVec &vecCalY = m_outputWS->dataY(CALDATAINDEX);
-  MantidVec &vecDiff = m_outputWS->dataY(DATADIFFINDEX);
-  MantidVec &vecCalPurePeak = m_outputWS->dataY(CALPUREPEAKINDEX);
-  MantidVec &vecCalBkgd = m_outputWS->dataY(CALBKGDINDEX);
+  auto &vecCalY = m_outputWS->mutableY(CALDATAINDEX);
+  auto &vecDiff = m_outputWS->mutableY(DATADIFFINDEX);
+  auto &vecCalPurePeak = m_outputWS->mutableY(CALPUREPEAKINDEX);
+  auto &vecCalBkgd = m_outputWS->mutableY(CALBKGDINDEX);
   for (size_t i = 0; i < numpts; ++i) {
     // Calculated (refined) data
     vecCalY[i] = vecCalPurePeaks[i] + vecBkgd[i];
```

Comment:
There are `operator=` and `operator+` etc. for the types in `HistogramData`, so you can avoid the loop.